include/fmt/compile.h

```diff
@@ -37,6 +43,27 @@ struct is_compiled_string : std::is_base_of<compiled_string, S> {};
  */
 #define FMT_COMPILE(s) FMT_STRING_IMPL(s, fmt::detail::compiled_string)
 
+#ifdef FMT_USE_NONTYPE_TEMPLATE_PARAMETERS
+template <typename Char, size_t N> struct fixed_string {
```

Comment:
Is `fixed_string` actually needed? Can we directly use an array instead?

Reply:
Looks like yes, it's actually needed. The raw C-style array should be wrapped in some type, and this wrapper-type should also have a constructor with `const Char (&str)[N]` as an argument. Of course, I can be wrong here, but all my attempts ended up with nothing working.

**But** I realized that the deduction guide for `fixed_string` can be removed now because it holds a null-terminated string.